fw/fw/multicast.go

```diff
@@ -61,26 +61,37 @@ func (s *Multicast) AfterReceiveInterest(
 	packet *defn.Pkt,
 	pitEntry table.PitEntry,
 	inFace uint64,
-	nexthops []*table.FibNextHopEntry,
+	nexthops [defn.MaxNextHops]*table.FibNextHopEntry,
+	nexthopsCount int,
 ) {
-	if len(nexthops) == 0 {
-		core.LogDebug(s, "AfterReceiveInterest: No nexthop for Interest=", packet.Name, " - DROP")
+	if nexthopsCount == 0 {
+		if core.HasDebugLogs() {
+			core.LogDebug(s, "AfterReceiveInterest: No nexthop for Interest=", packet.Name, " - DROP")
+		}
 		return
 	}
 
 	// If there is an out record less than suppression interval ago, drop the
 	// retransmission to suppress it (only if the nonce is different)
 	for _, outRecord := range pitEntry.OutRecords() {
-		if outRecord.LatestNonce != *packet.L3.Interest.NonceV &&
+		if outRecord.LatestNonce != *packet.L3.Interest.Nonce() &&
 			outRecord.LatestTimestamp.Add(MulticastSuppressionTime).After(time.Now()) {
-			core.LogDebug(s, "AfterReceiveInterest: Suppressed Interest=", packet.Name, " - DROP")
+			if core.HasDebugLogs() {
+				core.LogDebug(s, "AfterReceiveInterest: Suppressed Interest=", packet.Name, " - DROP")
+			}
 			return
 		}
 	}
 
 	// Send interest to all nexthops
-	for _, nexthop := range nexthops {
-		core.LogTrace(s, "AfterReceiveInterest: Forwarding Interest=", packet.Name, " to FaceID=", nexthop.Nexthop)
+	for i := 0; i < nexthopsCount; i++ {
```

Comment:
Why this is better than a range?

Reply:
Not very different, the alternative is to check the first nil value (`nexthops` is a fixed-size stack array). I just felt this is easier to read.